### PR TITLE
Roe 2276 fix dependency check issues

### PR DESCRIPTION
**Pull request by pjames-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2276?atlOrigin=eyJpIjoiNGNhN2YwZDNhZWMxNDUzOWFiNjNhOGEzZTJkZjA1NzUiLCJwIjoiaiJ9

### Change description

Updated spring libraries and json libraries to fix issues raised by dependency check. This fixes the follow issues:

- logback-core-1.2.7.jar: CVE-2021-42550(6.6)
- spring-web-5.3.13.jar: CVE-2021-22060(4.3), CVE-2022-22968(5.3), CVE-2023-20863(6.5), CVE-2023-20861(6.5), CVE-2023-20860(7.5), CVE-2022-22965(9.8), CVE-2022-22971(6.5), CVE-2022-22950(6.5), CVE-2022-22970(5.3)
- spring-boot-2.6.1.jar: CVE-2023-20883(7.5)
- spring-boot-starter-web-2.6.1.jar: CVE-2023-20883(7.5)
- spring-core-5.3.13.jar: CVE-2021-22060(4.3), CVE-2022-22968(5.3), CVE-2023-20863(6.5), CVE-2023-20861(6.5), CVE-2023-20860(7.5), CVE-2022-22965(9.8), CVE-2022-22971(6.5), CVE-2022-22950(6.5), CVE-2022-22970(5.3)
- spring-webmvc-5.3.13.jar: CVE-2021-22060(4.3), CVE-2022-22968(5.3), CVE-2023-20863(6.5), CVE-2023-20861(6.5), CVE-2023-20860(7.5), CVE-2022-22965(9.8), CVE-2022-22971(6.5), CVE-2022-22950(6.5), CVE-2022-22970(5.3)

**Note:** the json library is still (even the current version) vulnerable to CVE-2022-45688(7.5) and spring-web is still vulnerable to CVE-2016-1000027(9.8). 

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by mattch1 — APPROVED** [review]
Advise getting  Martin to approve as well.

File: pom.xml [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
     <log4j.version>2.17.2</log4j.version>
     <avro.version>1.11.1</avro.version>
     <ch-kafka.version>1.4.8</ch-kafka.version>
     <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
     <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
     <structured-logging.version>1.9.19</structured-logging.version>
-    <api-sdk-manager-java-library.version>1.0.4</api-sdk-manager-java-library.version>
-    <json.version>20230227</json.version>
+    <api-sdk-manager-java-library.version>1.0.7</api-sdk-manager-java-library.version>
+    <json.version>20230618</json.version>
```
Comment (mwestacott) [inline pom.xml:23]: The JSON CVE still seems to be present but commit message suggests it should be gone:

`json-20230618.jar (pkg:maven/org.json/json@20230618, cpe:2.3:a:json-java_project:json-java:*:*:*:*:*:*:*:*) : CVE-2022-45688`

File: pom.xml [hunk 2]
```diff
@@ -11,16 +11,16 @@
   <name>auth-code-notification</name>
   <properties>
     <java.version>11</java.version>
-    <spring-boot-dependencies.version>2.6.1</spring-boot-dependencies.version>
-    <spring-boot-maven-plugin.version>2.6.1</spring-boot-maven-plugin.version>
+    <spring-boot-dependencies.version>2.6.15</spring-boot-dependencies.version>
```
Comment (mwestacott) [inline pom.xml:14]: Some Spring vulnerabilities are still being reported. Was the intention of this PR to address them all?

Comment (pjames-ch) [reply pom.xml:14]: Not all - the ones addressed are in spring-core and one in spring-web, but not all in spring- web.

File: pom.xml (hunk 1 above)
Comment (pjames-ch) [reply pom.xml:23]: Yes - apparently that one is still open with the library, see https://nvd.nist.gov/vuln/detail/CVE-2022-45688. Shall I update commit message, or supress this in another commit and note it?

Comment (mwestacott) [reply pom.xml:23]: I think PR commit message/description is now fine. Clears this up for me. Thanks.

**Pull request merged**